juju-dhx

```diff
@@ -183,11 +203,13 @@ def do_import_ids(unit, import_ids, opts):
         return
     env = get_env(opts)
     machine = unit.get('Machine')
-    annotations = env.get_annotation(machine, 'machine')
-    if annotations['Annotations'].get('import-ids') != ','.join(import_ids):
+    annotations = env.get_annotation(machine, 'machine')['Annotations'] or {}
+    import_ids = ['lp:{}'.format(iid) if ':' not in iid else iid
```

Comment:
this line makes my head explode. :bomb:  I was looking for where iid is declared, and its on 208. Using a variable before assignment... oh my lambda.